staging/src/kubevirt.io/api/clone/v1alpha1/types.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VirtualMachineClone is a CRD that clones one VM into another.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+type VirtualMachineClone struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualMachineCloneSpec   `json:"spec" valid:"required"`
+	Status VirtualMachineCloneStatus `json:"status,omitempty"`
+}
+
+type CloneIncludeExclude struct {
+	// +optional
+	CloneAllByDefault *bool `json:"cloneAllByDefault,omitempty"`
+	// +optional
+	// +listType=atomic
+	Exclude []string `json:"exclude,omitempty"`
+	// +optional
+	// +listType=atomic
+	Include []string `json:"include,omitempty"`
+}
+
+type ObjectReference struct {
+	corev1.TypedLocalObjectReference `json:",inline"`
+	Namespace                        string `json:"namespace"`
+}
+
+type VirtualMachineCloneSpec struct {
+	Source *ObjectReference `json:"source"`
+
+	// +optional
+	Target *ObjectReference `json:"target,omitempty"`
```

Comment:
Should not be able to specify different target namespace.

Reply:
That's right. See my comment below. If we stick with the current approach I will definitely add it.